src/services/refactors/extractMethod.ts

```diff
@@ -921,7 +1022,42 @@ namespace ts.refactor.extractMethod {
 
         return { target, usagesPerScope, errorsPerScope };
 
+        function hasTypeParameters(node: Node) {
+            return isDeclarationWithTypeParameters(node) &&
+                node.typeParameters !== undefined &&
+                node.typeParameters.length > 0;
+        }
+
+        function isInGenericContext(node: Node) {
+            for (; node; node = node.parent) {
+                if (hasTypeParameters(node)) {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        function recordTypeParameterUsages(type: Type) {
+            // PERF: This is potentially very expensive.  `type` could be a library type with

```

Comment:
>This is potentially very expensive [](start = 21, length = 34)

Is this a good point to check for cancellation then?

Reply:
I think cancellation would need to be pushed down into the walker to make it useful.  That's a big enough change that I'd prefer to make it separately.